### ISSUE TITLE
Add support for Unicode normalization output iterators

### ISSUE THREAD

**Issue by tahonermann** [tahonermann/text_view#20]

**Comment by ruoso** [comment]
I would argue that normalization shouldn't be an output iterator, but rather a transform algorithm.

**Comment by tahonermann** [comment]
I tend to agree that, in most cases, treating normalization as a transcoding operation is probably what is desirable.  Such an interface can be specialized for particular iterators (pointers) to provide higher performance as well.  The benefit of an output iterator (or a proxy input iterator) is that the transformation can be done lazily.  I think both interfaces have their uses.

**Comment by ruoso** [comment]
I think that is an academical use case. In practice the reason why you need
to normalize the text is because you will perform an operation that
requires the text to be normalized.

Making the normalizing into an output iterator would be technically valid,
but semantically confusing.

Em sáb, 24 de set de 2016 18:14, Tom Honermann <notifications@github.com>
escreveu:

**Comment by tahonermann** [comment]
I can see use cases for wanting to perform an operation on normalized text in a lazy fashion.

Regardless, I agree that a transcoding interface has more potential uses.  That falls under issue #4.
